src/components/Filters/FiltersWithRedux.jsx

```diff
@@ -0,0 +1,214 @@
+import "./Filters.css";
+import { Gridd } from "../Grid/Grid";
+import { useEffect } from "react";
+import { useDispatch, getState } from "react-redux";
+
+
+import {
+  filterByCategory,
+  filterByRating,
+  getAllData,
+  filterLowtoHigh,
+  filterHightoLow,
+} from "../../redux/Slice";
+
+
+const Axios = require("axios");
+
+export const FilterWithRedux = () => {
+  const dispatch = useDispatch();
+
+  //Functions to get data acc. to clicked filter
+
+  useEffect(() => {
+    //Get Default Data
+    const showData = () => {
+      Axios.get("http://localhost:3001/data").then((res) => {
+        dispatch(getAllData(res.data));
+      });
+    };
+    showData();
+  }, [dispatch]);
```

Comment:
```suggestion
dispatch(getAllData(res.data));

This is incorrect usage of useEffect
```

Reply:
Instead of defining showData function inside useEffect, i have defined the function outside and calling this showData function inside use effect, with a dependency on dispatch. I believe this might be the correct implementation